### PR TITLE
fix(script): update modifiedBy/modifiedOn project fields.

### LINKED ISSUE #2316: fix(script): update modifiedBy/modifiedOn project fields.
[//]: # (Copyright Siemens AG, 2021. Part of the SW360 Portal Project)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

### Description

Getting 500 error (internal server error ) while hitting the Summa Administration(resource/api/projects/{id}/summary/administration) rest Api. Because many of projects doesn't have modifiedBy and modifiedBy fields in project table.

Please describe your situation in few words here.

#### Steps followed and expected result

Describe the steps followed by you and your expected results after following the steps.

### PR DESCRIPTION
**Pull request by nikkuma7**
[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.
> * Which issue is this pull request belonging to and how is it solving it? (*Refer to issue here*)
> * Did you add or update any new dependencies that are required for your change?

Issue: Closes #2316 

### How To Test?
resource/api/projects/{id}/summary/administration
ex: resource/api/projects/c109935d14e846c4afe03a8f3e97798b/summary/administration

will not give 500 error(internal server error)

> How should these changes be tested by the reviewer?
> Have you implemented any additional tests?

### Checklist
Must:
- [ ] All related issues are referenced in commit messages and in PR

### CODE REVIEW

**Comment by rudra-superrr** [comment]
change the filename to 061_add_modifiedBy_modifiedOn_project.py and add it in README.md file(inside migration folder only).

**Comment by nikkuma7** [comment]
> change the filename to 061_add_modifiedBy_modifiedOn_project.py and add it in README.md file(inside migration folder only).

changes done.

**Review by rudra-superrr — CHANGES_REQUESTED** [review]

File: scripts/migrations/61_add_modifiedBy_modifiedOn_project.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+def run():
+    logFile = open('f53_add_modifiedBy_modifiedOn_project.py', 'w')
+    all_projects = db.find(get_all_project);
+    all_ = list(all_projects)
+    print ('found ' + str(len(all_)) + ' projects')
+    log = {}
+   
+
+    for prj in all_:
+        if prj.get("modifiedBy") is None and prj.get("modifiedOn") is None:
```
[the remark this answers is not in the document]
Comment (nikkuma7) [reply scripts/migrations/61_add_modifiedBy_modifiedOn_project.py]: done

File: scripts/migrations/61_add_modifiedBy_modifiedOn_project.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            prj[MODIFIED_BY]=""
+            prj[MODIFIED_ON] ="" 
+            db.save(prj);
+    json.dump(log, logFile, indent = 4, sort_keys = True)
+    logFile.close()
+
+    print ('\n')
+    print ('------------------------------------------')
+    print ('Please check log file "61_add_modifiedBy_modifiedOn_project.py" in this directory for details')
```
[the remark this answers is not in the document]
Comment (nikkuma7) [reply scripts/migrations/61_add_modifiedBy_modifiedOn_project.py]: done

File: scripts/migrations/61_add_modifiedBy_modifiedOn_project.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+# get all projects
+get_all_project = {"selector": { "type": { "$eq": "project" }}, "limit": 3000}
+
+
+# ---------------------------------------
+# functions
+# ---------------------------------------
+
+def run():
+    logFile = open('f53_add_modifiedBy_modifiedOn_project.py', 'w')
```
[the remark this answers is not in the document]
Comment (nikkuma7) [reply scripts/migrations/61_add_modifiedBy_modifiedOn_project.py]: done

File: scripts/migrations/61_add_modifiedBy_modifiedOn_project.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+MODIFIED_BY= 'modifiedBy'
+MODIFIED_ON= 'modifiedOn'
+
+
+# ----------------------------------------
+# queries
+# ----------------------------------------
+
+# get all projects
+get_all_project = {"selector": { "type": { "$eq": "project" }}, "limit": 3000}
```
[the remark this answers is not in the document]
Comment (nikkuma7) [reply scripts/migrations/61_add_modifiedBy_modifiedOn_project.py]: done

File: scripts/migrations/61_add_modifiedBy_modifiedOn_project.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+# SPDX-License-Identifier: EPL-2.0
+#
+# This is a manual database migration script. It is assumed that a
+# dedicated framework for automatic migration will be written in the
+# future. When that happens, this script should be refactored to conform
+# to the framework's prerequisites to be run by the framework. For
+# example, server address and db name should be parameterized, the code
+# reorganized into a single class or function, etc.
+#
+# This script is to fix corrupted releases.
```
[the remark this answers is not in the document]
Comment (nikkuma7) [reply scripts/migrations/61_add_modifiedBy_modifiedOn_project.py]: done

File: scripts/migrations/061_add_modifiedBy_modifiedOn_project.py [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+# get all projects
+get_all_project = {"selector": {"type": { "$eq": "project" },"modifiedOn": { "$exists": False },"modifiedBy": { "$exists": False }},"limit": 99999}
+
+
+# ---------------------------------------
+# functions
+# ---------------------------------------
+
+def run():
+    logFile = open('061_add_modifiedBy_modifiedOn_project.py', 'w')
```
Comment (rudra-superrr) [inline scripts/migrations/061_add_modifiedBy_modifiedOn_project.py:54]: logFile extension should be .log and not .py

Comment (nikkuma7) [reply scripts/migrations/061_add_modifiedBy_modifiedOn_project.py]: done.

File: scripts/migrations/061_add_modifiedBy_modifiedOn_project.py [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+    all_projects = db.find(get_all_project);
+    all_ = list(all_projects)
+    print ('found ' + str(len(all_)) + ' projects')
+    log = {}
+   
+
+    for prj in all_:
+           prj[MODIFIED_BY]=""
+           prj[MODIFIED_ON] =""
+           db.save(prj);
```
Comment (rudra-superrr) [inline scripts/migrations/061_add_modifiedBy_modifiedOn_project.py:64]: where is if condition for DRY_RUN?

**Comment by rudra-superrr** [comment]
logFile is empty.

image

**Comment by rudra-superrr** [comment]
changes look good and testing was successful.

**Pull request merged**